pkg/operator/ceph/csi/template/cephfs/csi-cephfsplugin-provisioner-dep.yaml

```
@@ -114,6 +114,7 @@ spec:
             - "--forcecephkernelclient={{ .ForceCephFSKernelClient }}"
             - "--metricspath=/metrics"
             - "--enablegrpcmetrics={{ .EnableCSIGRPCMetrics }}"
+            - "--setmetadata={{ .CSIEnableMetadata }}"
```

should this only be set when the option is enabled? Older Ceph-CSI versions do not have this commandline option.

you probably need to mention somewhere what Ceph-CSI version added this option

---

Will CSI 3.7 be required for this option?

---

Yes, once the Ceph-CSI 3.7 is out, we will send a PR to  set the minimum required version to 3.7

---

Minimum required version just for this flag? Or minimum version for all of rook? We just need more overlap of versions before we set the min version for all of rook.

---

The minimum version for all rook https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/version.go#L29 ?

I have not seen anything that uses min CSI version check per flag so far!!

---

how about the following ?
```suggestion
            {{ if .CSIEnableMetadata }}
            - "--setmetadata={{ .CSIEnableMetadata }}"
            {{ end }}           
```

---

Whether `CSIEnableMetadata` is true or false we need to pass `--setmetadata`, because in the future if we are enabling metadata by default at the CSI level and incase a user wants to disable it, then this will not allow.

---

> The minimum version for all rook https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/version.go#L29 ?
> 
> I have not seen anything that uses min CSI version check per flag so far!!

If the CSI version is not the minimum (currently 3.5), rook will fail the reconcile [here](https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/version.go#L63-L66).

---

> Whether `CSIEnableMetadata` is true or false we need to pass `--setmetadata`, because in the future if we are enabling metadata by default at the CSI level and incase a user wants to disable it, then this will not allow.

We could potentially bump the min version to 3.6, but we can't immediately bump the min version to 3.7, otherwise it can be disruptive for users who rely on some version overlap for time to adjust their deployments. So we need to make the new flag optional

---

Sure, make sense.

@Madhu-1 for the time being we will have to make it optional, looks like we don't have a choice.

---

@travisn In the meanwhile, I see the minimum version is now set to CephCSIVersion{3, 7, 0} https://github.com/rook/rook/blob/master/pkg/operator/ceph/csi/version.go#L29

Will it make sense to mandate `--setmetadata` flag now?

---

Yes, i set it to 3.7.0 as a minimum by default we can keep this `if check` also so anyone can still use cephcsi 3.6.0 with AllowUnsupported flag.

---

ok, done now.